crates/spirv-std/src/lib.rs

```diff
@@ -51,6 +51,26 @@ pub use glam;
 pub use num_traits;
 pub use textures::*;
 
+/// Calls the `OpDemoteToHelperInvocationEXT` instruction, which corresponds to discard() in HLSL
+pub fn demote_to_helper_invocation() {
+    #[cfg(target_arch = "spirv")]
+    unsafe {
+        asm!(
+            "OpExtension \"SPV_EXT_demote_to_helper_invocation\"",
+            "OpCapability DemoteToHelperInvocationEXT",
+            "OpDemoteToHelperInvocationEXT"
+        );
+    }
+}
+
+/// Calls the `OpKill` instruction, which corresponds to discard() in GLSL
+pub fn discard() {
```

Comment:
To be consistent with the other function, should we have this one called `kill` so that it matches the convention of name of instruction except `Op`.
```suggestion
pub fn kill() {
```

Reply:
I want to match user's expectations of having a function/intrinsic called `discard()`. Not having it seems problematic, and causes unnecessary friction - blindly copying the names of SPIR-V things is something we'd like to avoid (@Jasper-Bekkers believes this much stronger than I, haha). I chose to default the `discard()` implementation to the GLSL one instead of the HLSL one because it's supported by default instead of requiring an extension (see error message).

Reply:
Something we could do is have a
```rust
mod glsl { fn discard(); }
mod hlsl { fn discard(); }
fn discard() { glsl::discard(); }
```
not sure if that's better/worse

Reply:
> blindly copying the names of SPIR-V things is something we'd like to avoid

I mean this is ostensibly this is `spirv-std` so I think matching their naming makes more sense rather than having a mix of semantics from different specs and languages.

If discoverability is a concern, we can add a doc alias so that when users search for `discard` they find this function. E.g.

```rust
#[doc(alias = "discard")]
fn kill() {}
```